### PR TITLE
NEW Allow to set a default project for POS sales

### PR DESCRIPTION
**Pull request by ymollard**
# NEW Allow to set a default project for POS sales
The TakePOS module can be set to a default project in the admin settings, so that all future sales are affected to this project.

### CODE REVIEW

File: htdocs/takepos/invoice.php [hunk 1]
```diff
@@ -149,6 +150,11 @@ function fail($message)
 	$soc->fetch(getDolGlobalString($constforcompanyid));
 }
 
+// Assign a default project, if relevant
+if (isModEnabled('project') && !empty(getDolGlobalString("CASHDESK_ID_PROJECT".$takeposterminal))) {
```
Comment (eldy) [inline htdocs/takepos/invoice.php:154]: !empty(getDolGlobalString(...  is always true.
I think you want to do 
!getDolGlobalString(...

Comment (ymollard) [reply htdocs/takepos/invoice.php:154]: I want to execute that piece of code only if the global string exists and is not an empty string.
*  If it's "some_string", then getDolGlobalString() will return that string, so empty() will be false, so !empty() will be true and will execute the code
* If it's "", then getDolGlobalString() will return "", so empty() will be true, so !empty() will be false and will not execute the code
* It it's not set, then getDolGlobalString() will return "", and the latest behaviour applies

Where am I mistaken?

Comment (eldy) [reply htdocs/takepos/invoice.php:154]: I made a mistake. The !empty(getDolGlobalString() will not be always true.
Bit the call of empty is useless, this is why the cti return an error expecting a more simple condition.
Test 
If (getDolGlobalString())
Is enough.

Comment (ymollard) [reply htdocs/takepos/invoice.php]: Fixed with getDolGlobalInt() instead

Comment (ymollard) [reply htdocs/takepos/invoice.php]: Poke (bis) @eldy

**Comment by ymollard** [comment]
Can this PR now be merged, @eldy?

**Pull request merged**